internal/antewrapper/tx_gas_limit_decorator.go

```diff
@@ -0,0 +1,39 @@
+package antewrapper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// TxGasLimitDecorator will check if the transaction's gas amount is higher than
+// 5% of the maximum gas allowed per block.
+// If gas is too high, decorator returns error and tx is rejected from mempool.
+// If gas is below the limit, then call next AnteHandler
+// CONTRACT: Tx must implement FeeTx to use TxGasLimitDecorator
+type TxGasLimitDecorator struct{}
+
+// MinTxPerBlock is used to determine the maximum amount of gas that any given transaction can use based on the block gas limit.
+const MinTxPerBlock = 20
+
+func NewTxGasLimitDecorator() TxGasLimitDecorator {
+	return TxGasLimitDecorator{}
+}
+
+func (mfd TxGasLimitDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+	}
+	// Ensure that the requested gas does not exceed the configured block maximum
+	gas := feeTx.GetGas()
+	gasTxLimit := uint64(0)
+	if ctx.BlockGasMeter() != nil {
+		gasTxLimit = ctx.BlockGasMeter().Limit() * (1 / MinTxPerBlock)
```

Comment:
1/20 will lead to 0.05 which maybe should be used, i am little worried about determinism of floating point division on gol-lang if say the MinTxPerBlock changes to 33.. i maybe overthinking this for this case but 🤷

Reply:
The current layout was intended to be as readable as possible and when I originally used the decimal value here it seemed less clear what we were targeting.  

As half the equation is a contstant (1/20) that part should be safe ... but we have a variable gas limit through config and that might cause issues if we used a non-evenly divisible value...

Reply:
> The current layout was intended to be as readable as possible and when I originally used the decimal value here it seemed less clear what we were targeting.
> 
> As half the equation is a contstant (1/20) that part should be safe ... but we have a variable gas limit through config and that might cause issues if we used a non-evenly divisible value...

i  updated the comment, i don't think ^^ calc works (unless i am missing something)

Reply:
Updatesd with a simplified expression works regardless of the limit specified.